core-kotlin-modules/core-kotlin-strings-2/src/test/kotlin/com/baeldung/whitespaces/RemoveWhitespaceTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.baeldung.whitespaces
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class RemoveWhitespaceTest {
+
+    @Test
+    fun givenString_whenReplace_WhitespacesShouldBeRemoved() {
+
+        val example = "House Of The Dragon"
+        val withOutSpaces = example.replace(" ", "")
+
+        assertThat(withOutSpaces).hasToString("HouseOfTheDragon")
+    }
+
+    @Test
+    fun givenString_whenReplaceWithRegex_WhitespacesShouldBeRemoved() {
+
+        val example = "Kotlin is a programming language"
+        val withOutSpaces = example.replace("\\s".toRegex(), "")
+
+        assertThat(withOutSpaces).hasToString("Kotlinisaprogramminglanguage")
+    }
+
+    @Test
+    fun givenString_whenFilterByNonWhitespace_WhitespacesShouldBeRemoved() {
+
+        val example = "House Of The Dragon"
+        val withOutSpaces = example.filter { !it.isWhitespace() }
```

Comment:
Mentioned this in the feedback as well, but maybe

```suggestion
        val withOutSpaces = example.filterNot { it.isWhitespace() }
```

would be clearer?